### ISSUE TITLE
Missing media on image and media fields.

### ISSUE THREAD

**Issue by Zasa-san** [huridocs/uwazi#4002]
Supporting files that are used in image or media fields can be deleted, leaving those fields with missing values:

To reproduce:
- Add a image/media field to a template
- Create an entity
- Add supporting image or video files
- Use those files in the image/media fields
- Save the entity
- Delete the supporting files
- Reload the page

This is somewhat related to the improvements being done in #2366

![image](https://user-images.githubusercontent.com/71732018/140808223-07f952d2-be0d-4f8c-88c6-31ec145b7c34.png)

**Comment by txau** [comment]
I hardly consider this a bug. If a user deletes a file that is being used somewhere as an image, it becomes a broken link.

**Comment by aphilop** [comment]
I disagree with the above assessment. Not informing at least the users about the consequences of their actions is bad usability.
In my opinion, we need a general and consistent approach for dealing with such situations across the application (e.g. in contrast we don't allow to delete a property if it is inherited in another template).

@juanmnl Definitely something to consider with the new Entity Viewer concept and the file management in particular.

**Comment by RafaPolit** [comment]
We are revisiting this issue in entity V2.  

We can remove the base problem by deleting the media properties that point to a file.  You deleted the file, those media fields no longer point to a broken file.

In the client / UI, there could be an additional check and this prompts a question: 'you are about to delete a file that is being used in media properties.  Do you want to continue?'.  Obviously we could create this validation in the backend but then having the API return an error that then the client needs to catch and then, upon confirmation, send a flag or something... but it's even more cumbersome.

I do agree with Jaume that this is "hardly a bug", but showing the broken links is, indeed, a bug in my opinion.  So at least cleaning after the deletion procedure is a good practice.

cc @aphilop @txau

**Comment by RafaPolit** [comment]
The instruction, as for now is:
Do nothing.  Don't fail validation, don't sanitize data, simply delete the file and leave the media broken.  Users can go in and save the sanitized metadata if they want.

Additional info we should have:  The media fields, when pointing to a local file, are they
- storing the file ID they are pointing to?
- storing the URL for the media field as a relative path to the image?
- storing an absolute path to the URL?


cc @Joao-vi

**Comment by Joao-vi** [comment]
> Additional info we should have: The media fields, when pointing to a local file, are they

storing the URL for the media field as a relative path to the image.

The exact URL pattern is as so: 

`value: "/api/files/{Here goes the filename}"`

The filename is automatically created by the system when the File gets uploaded. 

When a URL Attachment gets created, is different tho. 
The image/media property types stores the full URL of the external attachment, you can still delete the file from attachments and there's no  bug on UI, since we don't host that file, it's external, therefore you can still see the image/media normally.

@RafaPolit
